project/Secp256k1.scala

```diff
@@ -0,0 +1,53 @@
+import sbt._
+import Keys._
+
+import scala.sys.process._
+
+object Secp256k1 {
+
+  lazy val Secp256k1Config = config("secp256k1")
+  lazy val generate = taskKey[Unit]("Make sources")
+  lazy val secp256kOutputDir = settingKey[File]("Directory containing native library.")
+
+  //Clean shoud remove all directories created during the build process
+  //even better - it should reset the project to the HEAD state
+  //  git fetch origin
+  //  git reset --hard origin/master
+  //remove *.so *.dll and *.dylib files from resources folder
+  def cleanDirs(dirs: File*): Unit = dirs.foreach(dir => s"rm -rf $dir".!)
+
+  def cleanProject(projectDir: File, dirs: File*) = {
+    cleanDirs(dirs:_*)
+    val makeDistCleanCmd = s"make -C ${projectDir.getAbsolutePath} distclean"
+    makeDistCleanCmd.!
+  }
+
+  def secp256k1Generate(baseDir: File) = {
+    val autogenCmd: String = "./autogen.sh"
```

Comment:
So now we depend on `autotools`

Reply:
Yeah, that's what the build process requires.

Reply:
Do we need to have secp256k1 built during our build process? Can't we just add it to the prerequisites for the dev machine?